### DISCUSSION TITLE
Printer Template for HP Color LaserJet Pro MFP M479fdw

### DISCUSSION THREAD

**Discussion by larieu** [zabbix/community-templates#310] [Show and tell]
I have modified a template for HP OfficeJet 8xx to suit `HP Color LaserJet Pro MFP M479fdw` printer
[https://github.com/zabbix/community-templates/tree/main/Printers/HP/template_hp_officejet_pro_87xx](url)
( and reordered a bit more tidier ) 

>Tested on latest Zabbix (6.4) on a Rocky Linux 9 

the file is attached
I don't know how to add it into the list of templates
If anyone is able to do it; it might help some other people to monitor such a printer

Probably it is extremely easy to modify this template for other HP models
Steps to modify for other similar HP printers

try to gather SNMP info from your printer as this

`snmpwalk -v 2c -On -c your_community aaa.bbb.ccc.ddd .1.3.6.1.2.1.43 | grep "1.3.6.1.2.1.43.11.1.1.6.1.4\|1.3.6.1.2.1.43.11.1.1.6.1.1\|1.3.6.1.2.1.43.11.1.1.8.1.4\|1.3.6.1.2.1.43.11.1.1.9.1.4\|1.3.6.1.2.1.43.11.1.1.8.1.1\|1.3.6.1.2.1.43.11.1.1.9.1.1\|1.3.6.1.2.1.43.11.1.1.8.1.2\|1.3.6.1.2.1.43.11.1.1.9.1.2\|1.3.6.1.2.1.43.11.1.1.8.1.3\|1.3.6.1.2.1.43.11.1.1.9.1.3\|1.3.6.1.2.1.43.11.1.1.6.1.2\|1.3.6.1.2.1.25.3.2.1.3.1\|1.3.6.1.2.1.43.11.1.1.6.1.3"`

the result should look like this

> .1.3.6.1.2.1.43.11.1.1.6.1.1 = STRING: "Black Cartridge HP W2030A"
.1.3.6.1.2.1.43.11.1.1.6.1.2 = STRING: "Cyan Cartridge HP W2031A"
.1.3.6.1.2.1.43.11.1.1.6.1.3 = STRING: "Magenta Cartridge HP W2033A"
.1.3.6.1.2.1.43.11.1.1.6.1.4 = STRING: "Yellow Cartridge HP W2032A"
.1.3.6.1.2.1.43.11.1.1.8.1.1 = INTEGER: 100
.1.3.6.1.2.1.43.11.1.1.8.1.2 = INTEGER: 100
.1.3.6.1.2.1.43.11.1.1.8.1.3 = INTEGER: 100
.1.3.6.1.2.1.43.11.1.1.8.1.4 = INTEGER: 100
.1.3.6.1.2.1.43.11.1.1.9.1.1 = INTEGER: 40
.1.3.6.1.2.1.43.11.1.1.9.1.2 = INTEGER: 34
.1.3.6.1.2.1.43.11.1.1.9.1.3 = INTEGER: 33
.1.3.6.1.2.1.43.11.1.1.9.1.4 = INTEGER: 14

as you can spot the : 
`.1.3.6.1.2.1.43.11.1.1.6.1.x` is the name of the cartridge from the printer ( x - 1,2,3,4 )
`.1.3.6.1.2.1.43.11.1.1.8.1.x` is the MAX_LEVEL of the cartridge from the printer ( x - 1,2,3,4 )
`.1.3.6.1.2.1.43.11.1.1.9.1.x` is the LEFT_LEVEL of the cartridge from the printer ( x - 1,2,3,4 )

in the template you have to change the name for the cartridges as it is in your printer.(not mandatory - mandatory is only the mapping of the colors)
 
in the original template x=4 - black ; x=1 - cyan ; x=2 - magenta ; x=3 - yellow
where in my case is x=1 - black ; x=2 - cyan ; x=3 - magenta ; x=4 - yellow

and the name of the printer ( it is a simple replace) 

[template_hp_M479fdw.zip](https://github.com/zabbix/community-templates/files/13493091/template_hp_M479fdw.zip)

**Comment by woodsy900** [comment]
DUDE THANKYOU!
